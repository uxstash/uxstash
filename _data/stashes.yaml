- title: 80/20 rule
  category: Strategy
  description: "The Pareto principle (also known as the 80â€“20 rule, the law of the vital few, and the principle of factor sparsity) states that, for many events, roughly 80% of the effects come from 20% of the causes."
  url: http://northerndiv.com/improving-ux-80-20-rule/
  tag:

- title: Accessibility Analysis
  category: Development
  description: "A study to measure how the website is easy for everyone to use, regardless of their special needs. It could be done by specific tools, following the W3C guideline or even by an usability test."
  url: https://www.w3.org/wiki/Accessibility_testing
  tag:

- title: Adaptive Web Design (AWD)
  category: Development
  description: "Adaptive websites adapt to the width of the browser at a specific points. In other words, the website is only concerned about the browser being a specific width, at which point it adapts the layout."
  url: http://www.sitepoint.com/moving-beyond-responsive-web-adaptive-web/
  tag:

- title: Aesthetic-Usability Effect
  category: Design
  description: "The aesthetic usability effect is where a user will perceive an attractive product as easier to use than an ugly one."
  url: http://usabilityfriction.com/2010/03/30/aesthetic-usability-effect/
  tag:

# - title:
#   category:
#   description:
#   url:
#   tag:

# - title:
#   category:
#   description:
#   url:
#   tag:

# - title:
#   category:
#   description:
#   url:
#   tag:

# - title:
#   category:
#   description:
#   url:
#   tag:

# - title:
#   category:
#   description:
#   url:
#   tag:
