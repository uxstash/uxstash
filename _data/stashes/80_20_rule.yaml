stash: 80/20 rule
category: Strategy
description: "The Pareto principle (also known as the 80â€“20 rule, the law of the vital few, and the principle of factor sparsity) states that, for many events, roughly 80% of the effects come from 20% of the causes."
tag:
url:
  - http://northerndiv.com/improving-ux-80-20-rule/
  - http://www.webdesignerdepot.com/2011/02/the-8020-rule-applied-to-web-design/