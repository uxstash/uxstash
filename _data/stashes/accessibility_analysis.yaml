stash: Accessibility Analysis
category: Development
description: "A study to measure how the website is easy for everyone to use, regardless of their special needs. It could be done by specific tools, following the W3C guideline or even by an usability test."
tag:
article:
  - url: https://www.w3.org/wiki/Accessibility_testing
  - url: http://www.tutorialspoint.com/software_testing_dictionary/accessibility_testing.htm
  - url: http://jimthatcher.com/testing.htm