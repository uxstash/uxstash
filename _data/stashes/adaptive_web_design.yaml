stash: Adaptive Web Design (AWD)
category: Development
description: "Adaptive websites adapt to the width of the browser at a specific points. In other words, the website is only concerned about the browser being a specific width, at which point it adapts the layout."
tag:
article:
  - url: http://www.sitepoint.com/moving-beyond-responsive-web-adaptive-web/
    title:
    source:
