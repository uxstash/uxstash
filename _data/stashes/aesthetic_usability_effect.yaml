stash: Aesthetic-Usability Effect
category: Design
description: "The aesthetic usability effect is where a user will perceive an attractive product as easier to use than an ugly one."
tag:
article:
  - url: http://usabilityfriction.com/2010/03/30/aesthetic-usability-effect/
    title:
    source:
