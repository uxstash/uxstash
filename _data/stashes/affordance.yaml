stash: Affordance
category: Design
description: "A situation where an objectâ€™s sensory characteristics intuitively imply its functionality and use."
tag:
article:
  - url: http://www.uxbooth.com/articles/foundations-of-affordances/
    title:
    source:
