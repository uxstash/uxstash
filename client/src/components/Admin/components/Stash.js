import React from "react";
import { DropTarget } from "react-dnd";

const Types = {
  ITEM: "article"
};

const spec = {
  drop(props, monitor, component) {
    return props;
  }
};

function collect(connect, monitor) {
  return {
    connectDropTarget: connect.dropTarget()
  };
}

class Stash extends React.Component {
  render() {
    const { connectDropTarget, title } = this.props;
    return connectDropTarget(<div className="stash">{title}</div>);
  }
}

export default DropTarget(Types.ITEM, spec, collect)(Stash);
