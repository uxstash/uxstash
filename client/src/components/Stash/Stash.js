import React from 'react';

class Stash extends React.Component {
  render() {
    return (
      <main>
        <div className="container">
        
        </div>
      </main>
    )
  }
}

export default Stash;