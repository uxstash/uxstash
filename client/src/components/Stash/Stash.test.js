import React from "react";
import { shallow } from "enzyme";
import stashes from "../../data/stashes";

import Stash from "./Stash";

describe("Stash", () => {
  it("should render correctly", () => {
    const component = shallow(<Stash match={{ params: { id: "1" } }} />);
    expect(component).toMatchSnapshot();
  });

  it("should have the correct title, description, number of articles", () => {
    const component = shallow(<Stash match={{ params: { id: "1" } }} />);

    const title = stashes[0].title;
    const description = stashes[0].description;
    const articlesLength = stashes[0].articles.length;

    expect(component.find("h1").text()).toContain(title);
    expect(component.find("p.description").text()).toEqual(description);
    expect(component.find(".table-body a")).toHaveLength(articlesLength);
  });

  it("should have a working back button", () => {
    const component = shallow(<Stash match={{ params: { id: "1" } }} />);

    const category = stashes[0].category.toLowerCase();

    expect(component.find(".btn-back").props().to).toBe(`/${category}`);
  });
});
