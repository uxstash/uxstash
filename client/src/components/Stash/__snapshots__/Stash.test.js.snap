// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stash should render correctly 1`] = `
<main>
  <div
    className="container"
  >
    <h1>
      <Link
        className="btn-back"
        replace={false}
        to="/strategy"
      >
        <IoMdArrowBack />
      </Link>
      80/20 rule
    </h1>
    <p
      className="description"
    >
      The Pareto principle (also known as the 80â€“20 rule, the law of the vital few, and the principle of factor sparsity) states that, for many events, roughly 80% of the effects come from 20% of the causes.
    </p>
    <div
      className="table"
    >
      <div
        className="table-row table-header"
      >
        <div
          className="table-cell x2"
        >
          About
        </div>
        <div
          className="table-cell"
        >
          Source
        </div>
        <div
          className="table-cell"
        >
          Date
        </div>
      </div>
      <div
        className="table-row table-body"
        key="0"
      >
        <div
          className="table-cell x2"
        >
          <a
            href="http://www.webdesignerdepot.com/2011/02/the-8020-rule-applied-to-web-design/"
            rel="noopener noreferrer"
            target="_blank"
          >
            The 80/20 Rule Applied to Web Design
          </a>
        </div>
        <div
          className="table-cell"
        >
          WebdesignerDepot
        </div>
        <div
          className="table-cell"
        >
          date
        </div>
      </div>
      <div
        className="table-row table-body"
        key="1"
      >
        <div
          className="table-cell x2"
        >
          <a
            href="https://uxmag.com/articles/pareto-principle-based-user-research"
            rel="noopener noreferrer"
            target="_blank"
          >
            Pareto Principle-Based User Research
          </a>
        </div>
        <div
          className="table-cell"
        >
          UX Magazine
        </div>
        <div
          className="table-cell"
        >
          date
        </div>
      </div>
      <div
        className="table-row table-body"
        key="2"
      >
        <div
          className="table-cell x2"
        >
          <a
            href="http://www.keepitusable.com/blog/?tag=8020-rule"
            rel="noopener noreferrer"
            target="_blank"
          >
            Using the Pareto Principle to improve your user experience
          </a>
        </div>
        <div
          className="table-cell"
        >
          Keep It Usable
        </div>
        <div
          className="table-cell"
        >
          date
        </div>
      </div>
      <div
        className="table-row table-body"
        key="3"
      >
        <div
          className="table-cell x2"
        >
          <a
            href="http://www.measuringu.com/blog/pareto-ux.php"
            rel="noopener noreferrer"
            target="_blank"
          >
            Applying The Pareto Principle To The User Experience
          </a>
        </div>
        <div
          className="table-cell"
        >
          Measuring U
        </div>
        <div
          className="table-cell"
        >
          date
        </div>
      </div>
    </div>
  </div>
</main>
`;
